service/routers/users.js

```diff
@@ -0,0 +1,78 @@
+import { Router } from 'express'
+import { User } from '../models/User.js'
+
+const usersRouter = Router()
+
+function sanitizeUsers(users) {
+    const sanitizedUsers = users.map((user) => ({
+        id: user._id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        decks: user.decks,
+        active: user.active,
+        role: user.role
+    }))
+    return sanitizedUsers
+}
+
+const getUsers = async (req, res) => {
+  const { userId } = req.user
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
+    const users = await User.find({})
+    res.send(sanitizeUsers(users)) 
+  } else {
+    res.status(403).send('Forbidden')
+  }
+}
+
+const getUsersById = async (req, res) => {
+  const { userId } = req.user   
+  const requestor = await User.findById(userId)  
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
+    const user = await User.findById(req.params.id)
+    res.send(user) 
+  } else {
+    res.status(403).send('Forbidden')
+  }
+}
+
+const updateUser = async (req, res) => {
+  const { userId } = req.user 
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin') {
+    const result = await User.findByIdAndUpdate(req.params.id, req.body)
+    res.status(201).send(`User updated ${result}`)
+  } else if (requestor.role === 'superuser' || req.params.id === userId) {
```

Comment:
Any time you have repeated code you should consider refactoring. There's an easy way to improve this code.